spec/chrome_integration/gtalk.spec.js

```diff
@@ -0,0 +1,240 @@
+ *
+ * Possible flakiness in these tests may occur when using the default (shared)
+ * accounts if:
+ * - If multiple users run the tests at the same time, they may send messages
+ *   to each other, rather than between the 2 social clients running in the
+ *   same test
+ * - If tests are run too frequently, GTalk may throttle messages sooner
+ *   (return 503 "service unavailable")
+ * To prevent flakiness, these tests should only be run by 1 person at a time
+ * (per pair of tests accounts) with some time (rough estimate 10 minutes)
+ * between each attempt.
+ */
+
+
+var REDIRECT_URL = 'https://www.uproxy.org/oauth-redirect-uri';
+var CLIENT_ID =
+    '746567772449-jkm5q5hjqtpq5m9htg9kn0os8qphra4d.apps.googleusercontent.com';
+
+var OAuthView = function() {};
+OAuthView.prototype.initiateOAuth = function(redirectURIs, continuation) {
+  continuation({redirect: REDIRECT_URL, state: ''});
+  return true;
+};
+OAuthView.prototype.launchAuthFlow = function(authUrl, stateObj, continuation) {
+  if (!this.refreshToken) {
+    continuation(undefined, 'No refreshToken set.');
+    return;
+  }
+  return Helper.getAccessToken(this.refreshToken).then(function(accessToken) {
+    continuation(REDIRECT_URL + '?access_token=' + accessToken);
+  }).catch(function(e) {
+    continuation(undefined, 'Failed to get access token');
+  });
+};
+
+var Helper = {
+  // Returns a Promise that fulfills with an access token.
+  getAccessToken: function(refreshToken) {
+    return new Promise(function(fulfill, resolve) {
+      var data = 'refresh_token=' + refreshToken +
+          '&client_id=' + CLIENT_ID +
+          '&client_secret=h_hfPI4jvs9fgOgPweSBKnMu' +
+          '&grant_type=refresh_token';
+      var xhr = new XMLHttpRequest();
+      xhr.open('POST', 'https://www.googleapis.com/oauth2/v3/token', true);
+      xhr.setRequestHeader('content-type', 'application/x-www-form-urlencoded');
+      xhr.onload = function() {
+        fulfill(JSON.parse(this.response).access_token);
+      };
+      xhr.send(data);
+    });
+  },
+  // Returns a promise that fulfills when socialClient has been logged in
+  // using the loginEmail address.  loginEmail must have an associated
+  // refresh token.
+  loginAs: function(socialClient, loginEmail) {
+    return socialClient.login({
+        agent: 'integration',
+        version: '0.1',
+        url: '',
+        interactive: false,
+        rememberLogin: false
+      });
+  },
+  // Sets up an onClientState listener and invokes the callback function
+  // anytime a new client for the given userId appears as ONLINE.
+  onClientOnline: function(socialClient, userId, callback) {
+    var onlineClientIds = {};
+    socialClient.on('onClientState', function(clientState) {
+      if (clientState.userId == userId &&
+          clientState.status == 'ONLINE' &&
+          !onlineClientIds[clientState.clientId]) {
+        // Mark this client as online so we don't re-invoke the callback
```

Comment:
Can you clarify? Do we expect clientIds for both Alice and Bob to be different between tests?

One potential solution:
We should create an afterEach(...) clause that does necessary cleanup after every single test.
This would handle logging out of each socialClient.
onlineClientIds can be a property off of Helper, which would be cleared in the afterEach as well.

What do you think about that?